Source/CsDebugScript.Engine/UserType.cs

```diff
@@ -17,12 +17,26 @@ public static class VariableCastExtender
         /// <param name="variable">The variable.</param>
         public static T DynamicCastAs<T>(this Variable variable) where T : UserType
         {
-            if (variable == null || !variable.GetRuntimeType().Inherits<T>())
+            if (variable == null)
             {
                 return null;
             }
 
-            return variable.CastAs<T>();
+            CodeType runtimeType = variable.DereferencePointer().GetRuntimeType();
```

Comment:
DereferencePointer will throw exception if variable is not the pointer. GetRuntimeType won't work if variable is not the pointer. Are you calling this function with double pointer?